core/src/test/java/org/elasticsearch/index/mapper/fixed/FixedPointTests.java

```diff
@@ -0,0 +1,460 @@
+            .startObject("properties")
+            .startObject("field1").field("type", "fixed").field("decimal_places", 2).endObject()
+            .endObject()
+            .endObject().endObject().string();
+
+        client().admin().indices().preparePutMapping("test").setType("type").setSource(mapping).get();
+
+        client().prepareIndex("test", "type", "1")
+            .setSource("field1", 1.2345).setRefresh(true).execute().actionGet();
+
+        // Check 1 decimal places, should not match
+        SearchResponse search = client().prepareSearch()
+            .setQuery(termQuery("field1", 1.2))
+            .execute().actionGet();
+
+        assertHitCount(search, 0l);
+
+        // Check 2 decimal places
+        search = client().prepareSearch()
+            .setQuery(termQuery("field1", 1.23))
+            .execute().actionGet();
+
+        assertHitCount(search, 1l);
+        assertEquals(search.getHits().getAt(0).getSource().get("field1"), 1.2345);
+
+        // Check 3 decimal places -- should truncate down to 2
+        search = client().prepareSearch()
+            .setQuery(termQuery("field1", 1.239))
+            .execute().actionGet();
+
+        assertHitCount(search, 1l);
+        assertEquals(search.getHits().getAt(0).getSource().get("field1"), 1.2345);
+
+        // Check 4 decimal places -- should truncate down to 2
+        search = client().prepareSearch()
+            .setQuery(termQuery("field1", 1.2349))
+            .execute().actionGet();
+
+        assertHitCount(search, 1l);
+        assertEquals(search.getHits().getAt(0).getSource().get("field1"), 1.2345);
+
+
+        // Check ranges
+        search = client().prepareSearch()
+            .setQuery(rangeQuery("field1").gte(1.0).lte(2.0))
+            .execute().actionGet();
+
+        assertHitCount(search, 1l);
+        assertEquals(search.getHits().getAt(0).getSource().get("field1"), 1.2345);
+
+        search = client().prepareSearch()
+            .setQuery(rangeQuery("field1").gte(1.23).lte(2.0))
+            .execute().actionGet();
+
+        assertHitCount(search, 1l);
+        assertEquals(search.getHits().getAt(0).getSource().get("field1"), 1.2345);
+
+        search = client().prepareSearch()
+            .setQuery(rangeQuery("field1").gte(1.0).lte(1.23))
+            .execute().actionGet();
+
+        assertHitCount(search, 1l);
+        assertEquals(search.getHits().getAt(0).getSource().get("field1"), 1.2345);
+
+        search = client().prepareSearch()
+            .setQuery(rangeQuery("field1").gte(1.2399).lte(2.0))    // extra decimal places truncated
+            .execute().actionGet();
+
+        assertHitCount(search, 1l);
+        assertEquals(search.getHits().getAt(0).getSource().get("field1"), 1.2345);
+
+        search = client().prepareSearch()
+            .setQuery(rangeQuery("field1").gte(1.0).lte(1.2))    // not enough decimal places to match
+            .execute().actionGet();
+
+        assertHitCount(search, 0l);
+    }
+}
```

Comment:
Should we also have a test to ensure the fixed point numeric field works with aggregations?